components/service/glean/metrics.yaml

```diff
@@ -161,3 +161,56 @@ glean.internal.metrics:
       - https://bugzilla.mozilla.org/show_bug.cgi?id=1525045#c18
     notification_emails:
       - telemetry-client-dev@mozilla.com
+      
+glean.error:
```

Comment:
Mh, one thing just occurred to me: why can't we use `GleanError.OutOfRange["label"].add()" and instead have to use the `recordError` trick?

Reply:
Because the errors need to be reported in the pings that the metric that caused the error are sent on.  In other words, we need to use the metric's `sendInPings`, not `GleanError.OutOfRange`'s `sendInPings`.

Reply:
This is interesting. I was assuming we'd only get error metrics in the `metrics` ping, but you raise a good point. I think we didn't spec this, hence the confusion :) I feel like having errors in the same ping metrics live might make more sense, but I'm a bit concerned about its behaviour: say that metric `A` goes in 3 pings, and there's an error with that metric. Do we send a count of 1 with each ping? Moreover, if some error happens on a metric that goes in the `baseline` or `events` ping, we'll increase the ping size there (though it's negligible).

Reply:
> say that metric `A` goes in 3 pings, and there's an error with that metric. Do we send a count of 1 with each ping?

I'd say yes, with the added advantage that the error counters will be reset on the same cadence as the ping itself.  If we sent all errors in the metrics ping, we wouldn't know which ping timespan the errors happened in.  (Maybe that level of detail isn't useful, though).

But if the use case is just to collect some general health metrics on which metrics might have problems, it probably doesn't matter too much where we send them and having only one place to look for that might actually be an advantage.

Should we take this to a proposal?

Reply:
> 
> 
> > say that metric `A` goes in 3 pings, and there's an error with that metric. Do we send a count of 1 with each ping?
> 
> I'd say yes, with the added advantage that the error counters will be reset on the same cadence as the ping itself. If we sent all errors in the metrics ping, we wouldn't know which ping timespan the errors happened in. (Maybe that level of detail isn't useful, though).
> 
> But if the use case is just to collect some general health metrics on which metrics might have problems, it probably doesn't matter too much where we send them and having only one place to look for that might actually be an advantage.

Good catch, different advantages depending on who's the consumer of the data: if it's us, trying to answer the question "is the API for metric X error-prone?", maybe it makes sense to have everything in one place and just have the absolute counts. If it's for developers, trying to see "is my metric behaving correctly?" then it makes sense to have that in their metrics' pings.

> Should we take this to a proposal?

Yes, I think this is good material for a proposal.